includes/embeds/class-amp-youtube-embed-handler.php

```diff
@@ -77,6 +79,76 @@ public function unregister_embed() {
 		remove_filter( 'embed_oembed_html', [ $this, 'filter_embed_oembed_html' ], 10 );
 	}
 
+	/**
+	 * Sanitize HTML that are not added via Gutenberg.
+	 *
+	 * @param Document $dom  Document.
+	 *
+	 * @return void
+	 */
+	public function sanitize_raw_embeds( Document $dom ) {
+
+		$nodes = $dom->xpath->query( '//iframe[ contains( @src, "youtu" ) ]' );
```

Comment:
This XPath query could be made more strict. Currently this will match any `iframe[src]` that contains the string `youtu`, which means it would match URLs such as "https://mysite.com/blog/youtube-iz-da-best".

The iframe `src` used in the YouTube oEmbed ([for example](https://www.youtube.com/oembed?url=https://www.youtube.com/watch?v=q4xKvHANqjk)) begins with `https://www.youtube.com` URL, so we could ensure we only match that by doing something like:

```suggestion
		$nodes = $dom->xpath->query( '//iframe[ starts-with( @src, "https://www.youtube.com" ) ]' );
```

Reply:
Yes, the current XPath query will fetch all iframe with URL that contains "youtu" including the example that you gave.
But it won't be converted. Since, we also do domain checks before it.

As you mention, the code example that you gave will only fetch an iframe that begins with `https://www.youtube.com`.  that will not work with different versions of the youtube URL in an iframe.

Reply:
> But it won't be converted. Since, we also do domain checks before it.

Right, but rather than iterating through potential unwanted elements and checking to see if the `src` is correct, we could do that upfront with the XPath query.

> As you mention, the code example that you gave will only fetch an iframe that begins with https://www.youtube.com. that will not work with different versions of the youtube URL in an iframe.

Yes you're right, those variations can be added to the XPath query as well. I did something similar to that [here](https://github.com/ampproject/amp-wp/blob/0860f489b9e59357abb1c2941d13030b194b12be/includes/embeds/class-amp-youtube-embed-handler.php#L75-L82) but essentially it would be something like:


```php
$applicable_domains = [ 'youtu.be', 'youtube.com', 'youtube-nocookie.com' ];

$query_segments = array_map(
	static function ( $domain ) {
		return sprintf( 'starts-with( @src, "https://www.%1$s/" ) or starts-with( @src, "https://%1$s/" )', $domain );
	},
	$applicable_domains
);
$query          = implode( ' or ', $query_segments );

$nodes = $dom->xpath->query( sprintf( '//iframe[ %s ]', $query ) );
```

That should essentially get rid of having to do the domain check as it's now being done upfront.

Reply:
Thanks, @pierlon for the feedback. I have updated the PR. Please have a look.